### PR TITLE
Deprecating markers, annotations, fill, rectangles properties of the Plot class

### PR DESCRIPTION
**Pull request by Davide-sd**
#### References to other Issues or PRs

**NOTEs**:

1. this is the first PR related to Issue #23036 .
2. The first PRs (this one included) will address the necessary backwards incompatible changes.
3. The aforementioned issue will be used as a log with all PRs of the merging process.

#### Brief description of what is fixed or changed

The properties ``markers, annotations, fill, rectangles`` of the ``Plot`` class  (containing user-provided numerical data to be added on a plot) are deprecated. The new implementation saves user-provided numerical data into a new data series, `GenericDataSeries`, which can easily be processed by ``MatplotlibBackend``. This makes the properties unnecessary. It also means that user-provided numerical data is taken into consideration when combining multiple plots with the ``extend`` and ``append`` methods (previously, they were disregarded).

Instead of setting those properties directly, users should pass the homonym keyword arguments to the plotting functions.

**Motivation for this deprecation:** the implementation of the ``Plot`` class suggests that it is ok to add attributes and hard-coded if-statements in the ``MatplotlibBackend`` class to provide more and more functionalities for user-provided numerical data (e.g. adding horizontal lines, or vertical lines, or bar plots, etc). However, in doing so one would reinvent the wheel: plotting libraries already implement the necessary API. There is no need to hard code these things in this plotting module.

The plotting module should facilitate the visualization of symbolic expressions. Then, the best way to add custom numerical data is to retrieve the figure created by the plotting module and use the API of a particular plotting library. For example, with Matplotlib we would write:

```py
from sympy import *
var("x")
# visualize a symbolic expression
p = plot(cos(x))
# retrieve Matplotlib's figure and axes
fig, ax = p._backend.fig, p._backend.ax[0]
# add custom numerical data
ax.plot([0, 1, 2], [0, 1, -1], "*")
ax.axhline(0.5)
# visualize the figure
fig
```

#### Release Notes

* plotting
  * DEPRECATION: Deprecating `markers`, `annotations`, `fill`, `rectangles` properties of the `Plot` class. These can still be provided as keyword arguments to the `plot` function but setting the attributes on the returned `Plot` object is now deprecated.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
I haven't checked this very closely as I am not too familiar with any potential complex usage of the plotting module (using markers etc already goes beyond my own usage). That being said I think this broadly looks good.

The release note should say `DEPRECATION` rather than `BREAKING CHANGE` unless I have misunderstood something. The difference is that with `DEPRECATION` the old code continues to work but emits a warning. A `BREAKING CHANGE` means that the old code will not work any more. Typically the process is that something is deprecated and then later removed. At the point when t is removed that would be the breaking change.

The deprecation description in the docs could be a little clearer by giving an explicit example of old usage that is now considered deprecated and exactly what that specific example should be replaced by.

Also if the suggestion is to do `plot(..., markers=....)` is that something that would work in previous versions of sympy as well? If so  then it would be good to clearly say what style of code can be used both in older and newer versions of sympy without emitting any deprecation warning.

File: doc/src/explanation/active-deprecations.md [hunk 1]
```diff
@@ -76,6 +76,61 @@ SymPy deprecation warnings.
...
+hard code these things. The plotting module should facilitate the visualization
+of symbolic expressions. The best way to add custom numerical data is to
+retrieve the figure created by the plotting module and use the API of a
+particular plotting library. For example:
+
+```py
+# plot symbolic expression
+p = plot(cos(x))
+# retrieve Matplotlib's figure and axes object
+fig, ax = p._backend.fig, p._backend.ax[0]
```
Comment (sylee957) [inline doc/src/explanation/active-deprecations.md:125]: I don't think that it is good to encourage users to use implementation details like `_backend` though.
Maybe it is better to develop a public function, like `to_backend`, and encourage users that if you start to use backend directly, you should keep working with the backend.

Comment (Davide-sd) [reply doc/src/explanation/active-deprecations.md:125]: Agreed, that will come on a future PR!

**Comment by sylee957** [comment]
You need to fix the author tests

File: doc/src/explanation/active-deprecations.md [hunk 2]
```diff
@@ -76,6 +76,61 @@ SymPy deprecation warnings.
...
+### Deprecate markers, annotations, fill, rectangles of the Plot class
+The properties ``markers, annotations, fill, rectangles`` (containing
+user-provided numerical data to be added on a plot) are deprecated.
+The new implementation saves user-provided numerical data into appropriate
+data series, which can easily be processed by ``MatplotlibBackend``.
+Instead of setting those properties directly, users should pass the homonym
+keyword arguments to the plotting functions.
+
+The supported behavior is to pass keyword arguments to the plotting functions,
+which works fine both for previous and current implementation:
```
[the remark this answers is not in the document]
Comment (Davide-sd) [reply doc/src/explanation/active-deprecations.md]: Thanks, just resolved it.

**Comment by oscarbenjamin** [comment]
Looks good to me.

**Comment by oscarbenjamin** [comment]
@Davide-sd I edited the release note. Let me know if it looks incorrect.

This looks good so I'll merge it.

**Comment by Davide-sd** [comment]
Looks good, thank you!

**Pull request merged**